controllers/secret.go

```diff
@@ -34,19 +38,65 @@ const (
 
 // reconcileKubeConfigSecret will ensure that the kubeconfig Secret is present and up to date.
 func (r *OperatorPipelineReconciler) reconcileKubeConfigSecret(meta metav1.ObjectMeta) error {
+	pipeline := types.NamespacedName{
+		Namespace: meta.Namespace,
+		Name:      meta.Name,
+	}
+	instance := &certv1alpha1.OperatorPipeline{}
+	err := r.Client.Get(context.TODO(), pipeline, instance)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			log.Log.Info("pipeline resource not found. Ignoring since object must be deleted")
+			return nil
+		}
+		log.Log.Error(err, "unable to retrieve pipeline resource in namespace "+meta.Namespace)
+		return err
+	}
+	var secretName = KUBECONFIG_SECRET
+	if instance.Spec.KubeconfigSecretName != "" {
+		secretName = instance.Spec.KubeconfigSecretName
+	}
 	key := types.NamespacedName{
 		Namespace: meta.Namespace,
-		Name:      KUBECONFIG_SECRET,
+		Name:      secretName,
 	}
-
 	secret := newSecret(key)
 	if IsObjectFound(r.Client, key, secret) {
-		log.Log.Info("existing kubeconfig secret found")
+		log.Log.Info(fmt.Sprintf("existing %s secret found in namespace %s ", secretName, meta.Namespace))
+		kubeConfigSecret := &corev1.Secret{}
+		err := r.Client.Get(context.TODO(), key, kubeConfigSecret)
+		if err != nil {
+			log.Log.Error(err, "unable to get kubeconfig secret")
+			return err
+		}
+		if kubeConfigSecret.Data["kubeconfig"] == nil {
+			err = errors.New(fmt.Sprintf("the kubeconfig key in %s is empty!", secretName))
+			log.Log.Error(err, "The specifed kubeconfig secret does not contain a kubeconfig")
+			return err
+		}
+		config, err := clientcmd.RESTConfigFromKubeConfig(kubeConfigSecret.Data["kubeconfig"])
```

Comment:
I'm confused as to why we would create a new client, when we have r.Client already? Can controller-runtime not be used?

Reply:
The initial thinking was to excercise the provided kubeconfig to ensure it works.. We cant make the existing client use this other kubeconfig right ?

But as discussed on slack this probably should be removed as we said we should not be "validating" the secret beyond checking that it exists..

Reply:
You should still be able to use controller-runtime if you wanted to do that validation, but as you've said, yeah, as per the slack discussion, there's not a whole lot of value there.